source/ht_file.cpp

```diff
@@ -206,14 +182,16 @@ namespace Hatchit
         **/
         size_t File::Read(BYTE* out, size_t len)
         {
-            size_t _len = 0;
+            m_handle.read(reinterpret_cast<char*>(out), len);
+            size_t count = m_handle.gcount();
 
-            _len = fread(out, sizeof(BYTE), len, m_handle);
-
-            if (_len == 0)
-                throw FileException(m_path, errno);
+            //if (count == 0)
```

Comment:
We aren't throwing an exception anymore.  Either re-include this because consumers still expect an exception for the given conditions, or adjust the documentation to reflect the new behavior